cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -353,6 +354,26 @@ func StartControllers(s *options.CMServer, kubeClient *client.Client, kubeconfig
 		serviceaccountcontroller.DefaultServiceAccountsControllerOptions(),
 	).Run()
 
+	if s.HealthCheckControllerSyncPeriod > 0 {
+		config := client.AddUserAgent(kubeconfig, "healthchecker-controller")
+
+		// NB: in order for this health checker to work, the controller manager must be
+		// able to reach all of the API servers listed in the "default/kubernetes"
+		// endpoints.
+		checkerTransport, err := client.TransportFor(config)
+		if err != nil {
+			panic(err)
```

Comment:
This is essentially the same behavior as the `OrDie` functions used above (e.g. `clientset.NewForConfigOrDie` -- try to construct a new client, and `panic` if something fails, because we can't proceed without such fundamental operation (and probably don't want to proceed with launch).